src/SFA.DAS.EmployerAccounts/Queries/GetEmployerAgreement/GetEmployerAgreementQueryHandler.cs

```diff
@@ -94,5 +90,15 @@ public async Task<GetEmployerAgreementResponse> Handle(GetEmployerAgreementReque
                 LastSignedAgreement = lastSignedAgreement
             };
         }
+
+        private string GetUserFullName(string userId)
+        {
+            var externalUserId = Guid.Parse(userId);
+            var user = _database.Value.Users
+                .Where(m => m.Ref == externalUserId)
+                .Select(m => m)
```

Comment:
Why select the whole user, and then just return the name?  Could you not just select the name only?

Reply:
FullName is actually a combination of FirstName and LastName so EF can't defer that to the database.